### ISSUE TITLE
`mul_add` suggested for `no_std` where it is unavailable

### ISSUE THREAD

**Issue by tomgilligan** [rust-lang/rust-clippy#10728]
### Summary

I think the idea behind this lint is great.  Thank you for your work.  Unfortunately `mul_add` is not available with `no_std`.

`mul_add` should not be suggested for `no_std`

### Lint Name

suboptimal_flops

### Reproducer

I tried this code:

```rust
f64::from(temp_data_raw - 0x8000) * 7.8125e-3f64 - 256.0f64
```

I saw this happen:

```
warning: failed to automatically apply fixes suggested by rustc to crate `p1`

after fixes were automatically applied the compiler reported errors within these files:

  * p1/src/lib.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see 
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0599]: no method named `mul_add` found for type `f64` in the current scope
  --> p1/src/lib.rs:43:51
   |
43 |                 f64::from(temp_data_raw - 0x8000).mul_add(7.8125e-3f64, -256.0f64),
   |                                                   ^^^^^^^ method not found in `f64`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
Original diagnostics will follow.
```

I expected to see this happen:
No suggestion of `mul_add` for `no_std`

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: aarch64-unknown-linux-gnu
release: 1.69.0
LLVM version: 15.0.7
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by blyxyas** [comment]
Maybe this requires adding a new function to Clippy Utils or so. So it will take some time.
@rustbot claim
